compat/src/render.js

```diff
@@ -159,6 +159,14 @@ options.vnode = vnode => {
 			});
 		}
 
+		// Adding support for defaultValue in select tag
+		if (type == 'select' && normalizedProps.defaultValue) {
+			normalizedProps.value = toChildArray(props.children).forEach(child => {
+				child.props.selected =
+					normalizedProps.defaultValue.indexOf(child.props.value) != -1;
```

Comment:
`defaultValue` can be a non-array value too, like in the repro of the original reporter: https://stackblitz.com/edit/preact-bug-defaultvalue-zn7r9a .

Let's add that check here 👍

Reply:
Right now I have converted it to single select but multi select has issues. When I tried the following code

```javascript
		// Adding support for defaultValue in select tag
		if (type == 'select' && normalizedProps.defaultValue != null) {
			normalizedProps.value = toChildArray(props.children).forEach(child => {
				if (normalizedProps.multiple) {
					child.props.selected =
						normalizedProps.defaultValue.indexOf(child.props.value) != -1;
				} else {
					child.props.selected =
						normalizedProps.defaultValue == child.props.value;
				}
			});
		}
```

for multi selection only the last value is selected. any thoughts on why this is happening ?

Reply:
but I tried logging the value in the loop it was selected during iteration but some got overwritten during the next iteration.

Reply:
When I stopped the test from watch mode and ran it again it worked :)